lib/services/project-properties-service.ts

```diff
@@ -48,9 +48,41 @@ export class ProjectPropertiesService implements IProjectPropertiesService {
 		return updated;
 	}
 
-	public updateProjectProperty(projectData: IProjectData, mode: string, property: string, newValue: any) : IFuture<void> {
-		return (() => {
+	public removeProjectProperty(dataToBeUpdated: IProjectData, property: string, projectData?: IProjectData) : IProjectData {
+		let normalizedProperty = this.normalizePropertyName(property, projectData);
+		if(dataToBeUpdated) {
+			// TODO: Should we call onChanging in case the property is removed?
```

Comment:
I'm not sure, that's why I've left the comment. 
But with the current code, removeProjectProperty is called only when property is moved from configuration specific file to .abproject, so I think we should not call onChanging...